tests/test_series.py

```diff
@@ -3285,9 +3286,45 @@ def callable(x: int | NAType) -> str | NAType:
     check(
         assert_type(s.map(callable, na_action=None), "pd.Series[str]"), pd.Series, str
     )
+    check(assert_type(s.map(callable), "pd.Series[str]"), pd.Series, str)
 
     series = pd.Series(["a", "b", "c"])
     check(assert_type(s.map(series, na_action=None), "pd.Series[str]"), pd.Series, str)
+    check(assert_type(s.map(series), "pd.Series[str]"), pd.Series, str)
+
+    s2: pd.Series[float] = pd.Series([1.0, pd.NA, 3.0])
```

Comment:
We generally don't use the annotations in the tests.  So you can do
```python
s2 = pd.Series([1.0, pd.NA, 3.0], dtype=float)
# OR
s2 = pd.Series([1.0, pd.NA, 3.0]).astype(pd.Float64Dtype())
```
Then the type checkers should recognize `s2` as `Series[float]`

Note that internally, `pd.Series([1.0, pd.NA, 3.0])` has dtype `object`

Reply:
This first actually gives me an error `TypeError: float() argument must be a string or a real number, not 'NAType'`. While the second version does work, i cant get it to work for the integer case of `s = pd.Series([1, pd.NA, 3])`. While that does work at the construction point and for inferring `s: Series[int]`, the values that are passed into `callable` at `s.map(callable)` are actually floats.

Reply:
For `Series[int]`, you can use `s = pd.Series([1, pd.NA, 2], dtype="Int64")`

Sorry about my error on the `dtype=float`.

Reply:
```
# Step 1: Create the Series with pd.NA
s = pd.Series([1, pd.NA, 3], dtype="Int64")

print(s)
print(s.dtype)

# Define the callable function
def callable(x: int | NAType) -> str | NAType:
    print(x, type(x))
    if pd.isna(x):
        return pd.NA
    return str(x)

print(s)
print(s.map(callable, na_action=None))

```

actually still gives me this:

```
$ python testing.py
0       1
1    <NA>
2       3
dtype: Int64
Int64
0       1
1    <NA>
2       3
dtype: Int64
1.0 <class 'float'>
nan <class 'float'>
3.0 <class 'float'>
0     1.0
1    <NA>
2     3.0
dtype: object

```

Reply:
That's a known pandas issue.  See https://github.com/pandas-dev/pandas/issues/57189